File: 0125-valid-palindrome/0125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    //ok remove space and special characters
    const ans = s.replace(/[^A-Za-z0-9]/g,"").
    toLowerCase();
    const res = s.replace(/[^A-Za-z0-9]/g,"").
    toLowerCase().split("").reverse().join("");
   
    if(ans === res) return true;
    else return false;
};

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
       //string pall or not
    int n = s.length();
        //remove spaces and special characters
 s.erase(remove_if(s.begin(), s.end(), ::isspace), s.end());
  s.erase(remove_if(s.begin(), s.end(), [](char c) {
        return !isalnum(c);
    }), s.end());
        transform(s.begin(), s.end(), s.begin(), ::tolower);

        string t = s;
        reverse(begin(t),end(t));
        cout<<t<<" ";
        if(s==t){
            return true;
        }
        else return false;
}
};